src/main.rs

```diff
@@ -154,34 +238,48 @@ async fn main() -> Result<()> {
                 .collect::<Vec<_>>();
             println!("Groups found: {:?}", group_ids)
         }
-
-        CommandLineArgs::GroupAddMembers {
-            user_id,
-            group_id: group_id_string,
-            new_members: member_id_strings,
-        } => {
-            let sdk = initialize_sdk_from_file(&user_id).await?;
-            let group_id = GroupId::try_from(group_id_string)?;
-            let members_ids = member_id_strings
-                .iter()
-                .map(|u| UserId::try_from(u.as_str()))
-                .collect::<std::result::Result<Vec<_>, _>>()?;
-            let add_result = sdk.group_add_members(&group_id, &members_ids).await?;
-            let successes = add_result
-                .succeeded()
-                .iter()
-                .map(|user| user.id())
-                .collect::<Vec<_>>();
-            let failures = add_result
-                .failed()
-                .iter()
-                .map(|err| format!("User: {}, Error: {}", err.user().id(), err.error()))
-                .collect::<Vec<_>>();
-            println!("Successful adds: {:?}", successes);
-            println!("Failed adds: {:?}", failures);
-        }
     }
+    Ok(())
+}
 
+enum GroupMembershipFunction {
+    AddMembers,
+    AddAdmins,
+    RemoveMembers,
+    RemoveAdmins,
+}
+
+async fn change_group_membership(
+    user_id_string: String,
+    group_id_string: String,
+    user_strings: Vec<String>,
+    membership_function: GroupMembershipFunction,
+) -> Result<()> {
+    let sdk = initialize_sdk_from_file(&user_id_string).await?;
+    let group_id = GroupId::try_from(group_id_string)?;
+    let user_ids = user_strings
+        .iter()
+        .map(|u| UserId::try_from(u.as_str()))
+        .collect::<std::result::Result<Vec<_>, _>>()?;
+    let add_result = match membership_function {
```

Comment:
This actually turned out pretty well. maybe a different name for the value.

Reply:
Guess I missed that one. Fixed